runtime/runtime.go

```diff
@@ -161,6 +167,53 @@ func (r *interpreterRuntime) ExecuteScript(
 	return exportValue(value), nil
 }
 
+func scriptExecutionFunction(epSignature *sema.FunctionType, argumentCount int, arguments [][]byte, runtimeInterface Interface) func(inter *interpreter.Interpreter) (interpreter.Value, error) {
+	return func(inter *interpreter.Interpreter) (interpreter.Value, error) {
+		argumentValues := make([]interpreter.Value, argumentCount)
+
+		// Decode arguments against parameter types
+		for i, parameter := range epSignature.Parameters {
+			parameterType := parameter.TypeAnnotation.Type
+			argument := arguments[i]
+
+			exportedParameterType := exportType(parameterType)
+			var value cadence.Value
+			var err error
+
+			wrapPanic(func() {
+				value, err = runtimeInterface.DecodeArgument(
+					argument,
+					exportedParameterType,
+				)
+			})
+
+			if err != nil {
+				return nil, &InvalidScriptArgumentError{
+					Index: i,
+					Err:   err,
+				}
+			}
+
+			arg := importValue(value)
+
+			// Check that decoded value is a subtype of static parameter type
+			if !interpreter.IsSubType(arg.DynamicType(inter), parameterType) {
+				return nil, &InvalidScriptArgumentError{
+					Index: i,
+					Err: &InvalidTypeAssignmentError{
+						Value: arg,
+						Type:  parameterType,
+					},
+				}
+			}
+
+			argumentValues[i] = arg
+		}
```

Comment:
This is the same code as https://github.com/onflow/cadence/blob/a968f040503e86274f7ac9c522110ce66e99b110/runtime/runtime.go#L349-L386, which is correct, but maybe refactor both into a common function to ensure the logic stays the same for both entry points (transactions and scripts)